src/doc/en/developer/doctesting.rst

```diff
@@ -1338,9 +1343,267 @@ To give a json file storing the timings for each file, use the
+
+By default, ``./sage --fixdoctests`` runs the doctester and replaces the expected outputs
+of all examples by the actual outputs from the current version of Sage::
+
+    [mkoeppe@sage sage]$ ./sage --fixdoctests \
+                                --overwrite src/sage/arith/weird.py
+
+For example, when applied to this Python file::
+
+  | r"""
+  | ...
+  |
+  | EXAMPLES::
+  |
+  |     sage: 2 + 2
+  |     5
+  |     sage: factor("91")
+  |     "7" * "13"
+  | ...
+
+the doctest fixer edits the file as follows::
+
+  | r"""
+  | ...
+  |
+  | EXAMPLES::
+  |
+  |     sage: 2 + 2
+  |     4
+  |     sage: factor("91")
+  |     Traceback (most recent call last):
+  |     ...
+  |     TypeError: unable to factor '91'
+  | ...
+
+As this command edits the source file, it may be a good practice to first use ``git commit``
+to save any changes made in the file.
+
+After running the doctest fixer, it is a good idea to use ``git diff`` to check
+all edits that the automated tool made.
+
+An alternative to this workflow is to use the option ``--keep-both``. When expected and
+actual output of an example differ, it duplicates the example, marking the two copies
+``# optional - EXPECTED`` and ``# optional - GOT``. (Thus, when re-running the doctester,
+neither of the two copies is run; this makes ``./sage --fixdoctests`` idempotent.)
+
+When exceptions are expected by an example, it is standard practice to abbreviate
+the tracebacks using ``...``.  The doctest fixer uses this abbreviation automatically
+when formatting the actual output, as shown in the above example.
+To disable it so that the details of the exception
+can be inspected, use the option ``--full-tracebacks``. This is particularly useful
+in combination with ``--keep-both``::
+
+    [mkoeppe@sage sage]$ ./sage --fixdoctests --keep-both --full-tracebacks \
+                                --overwrite src/sage/arith/weird.py
+
+This will give the following result on the above example::
+
+  | r"""
+  | ...
+  |
+  | EXAMPLES::
+  |
+  |     sage: 2 + 2                                 # optional - EXPECTED
+  |     5
+  |     sage: 2 + 2                                 # optional - GOT
+  |     4
+  |     sage: factor("91")                          # optional - EXPECTED
+  |     "7" * "13"
+  |     sage: factor("91")                          # optional - GOT
+  |     Traceback (most recent call last):
+  |     ...
+  |     File "<doctest...>", line 1, in <module>
+  |     factor("91")
+  |     File ".../src/sage/arith/misc.py", line 2680, in factor
+  |     raise TypeError("unable to factor {!r}".format(n))
+  |     TypeError: unable to factor '91'
+  | ...
+  | """
+
+To make sure that all doctests are updated, you may have to use the option ``--long``::
+
+    [mkoeppe@sage sage]$ ./sage --fixdoctests --long \
+                                --overwrite src/sage/arith/weird.py
+
+If you are not comfortable with allowing this tool to edit your source files, you can use
```

Comment:
I get
```
$ ./sage --fixdoctests --long src/sage/coding/ag_code.py 
Running "sage -t src/sage/coding/ag_code.py"
No fixes made in 'src/sage/coding/ag_code.py'
```
Shouldn't it be ```Running "sage -t --long src/sage/coding/ag_code.py"```?

Reply:
Thanks for catching this! Fixed in a3e61ad4e4